skorch/callbacks/training.py

```diff
@@ -761,3 +762,64 @@ def on_train_end(self, net, **kwargs):
         self.checkpoint_.save_model(net)
         self.checkpoint_._sink("Final checkpoint triggered", net.verbose)
         return self
+
+
+class InputShapeSetter(Callback):
+    """Sets the input dimension of the PyTorch module to the input dimension
+    of the training data. This can be of use when using a skorch model within
+    a sklearn pipeline and grid-searching feature transformers (or using
+    feature selection methods).
```

Comment:
```suggestion
    """Sets the input dimension of the PyTorch module to the input dimension
    of the training data.
    
    This can be of use when the shape of X is not known beforehand,
    e.g. when using a skorch model within an sklearn pipeline and
    grid-searching feature transformers, or using feature selection
    methods.
```